pkg/utils/formfill.go

```diff
@@ -106,6 +125,78 @@ func FormInputFillSuggestions(inputs []FormInput) mapsutil.OrderedMap[string, st
 	return data
 }
 
+// FormSelectFill fills a map with selected values from a slice of FormSelect structs.
+// It iterates over each FormSelect struct in the inputs slice and checks for a selected option.
+// If a selected option is found, it adds the corresponding value to the map using the input's name as the key.
+// If no option is selected, it selects the first option and adds its value to the map.
+// The function returns the filled map.
+func FormSelectFill(inputs []FormSelect) mapsutil.OrderedMap[string, string] {
+	data := mapsutil.NewOrderedMap[string, string]()
+	for _, input := range inputs {
+		for _, option := range input.FormOptions {
+			if option.Selected != "" {
+				data.Set(input.Name, option.Value)
+				break
+			}
+		}
+
+		// If no option is selected, select the first one
+		if !data.Has(input.Name) && len(input.FormOptions) > 0 {
+			data.Set(input.Name, input.FormOptions[0].Value)
+		}
+	}
+	return data
+}
+
+// FormTextAreaFill fills the form text areas with placeholder values.
+// It takes a slice of FormTextArea structs as input and returns an OrderedMap
+// containing the form field names as keys and the placeholder values as values.
+func FormTextAreaFill(inputs []FormTextArea) mapsutil.OrderedMap[string, string] {
+	data := mapsutil.NewOrderedMap[string, string]()
+	for _, input := range inputs {
+		data.Set(input.Name, FormData.Placeholder)
+	}
+	return data
+}
+
+// FormFillSuggestions takes a slice of form fields and returns an ordered map
+// containing suggestions for filling those form fields. The function iterates
+// over each form field and based on its type, calls the corresponding fill
+// function to generate suggestions. The suggestions are then merged into a
+// single ordered map and returned.
+//
+// Parameters:
+// - formFields: A slice of form fields.
+//
+// Returns:
+// An ordered map containing suggestions for filling the form fields.
+func FormFillSuggestions(formFields []interface{}) mapsutil.OrderedMap[string, string] {
+	merged := mapsutil.NewOrderedMap[string, string]()
+	for _, item := range formFields {
+		switch v := item.(type) {
+		case FormInput:
+			dataMapInputs := FormInputFillSuggestions([]FormInput{v})
+			dataMapInputs.Iterate(func(key, value string) bool {
```

Comment:
almost identical to other instances, this can be a small func: func Merge(...) {...}

Reply:
You are right, i made a small function to merge OrderedMaps